### PR TITLE
MUIC-432: [Android] File download - Downloaded files are not shown on…

### PR DESCRIPTION
**Pull request by deniss-mc**
… the phone

MUIC-435: [Android] File download - Downloading a file and ending the engagement the file download screen still remains

### CODE REVIEW

File: widgetssdk/src/main/java/com/glia/widgets/filepreview/domain/usecase/PutImageFileToDownloadsUseCase.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package com.glia.widgets.filepreview.domain.usecase;
+
+import android.graphics.Bitmap;
+
+import com.glia.widgets.filepreview.data.GliaFileRepository;
+import com.glia.widgets.filepreview.domain.exception.FileNameMissingException;
+
+import io.reactivex.Completable;
+
```
Comment (mustafa-ogun-ozturk) [inline widgetssdk/src/main/java/com/glia/widgets/filepreview/domain/usecase/PutImageFileToDownloadsUseCase.java:9]: extra line

File: widgetssdk/src/main/res/values/themes.xml [hunk 2]
```diff
@@ -45,6 +45,10 @@
         <item name="android:backgroundDimEnabled">true</item>
     </style>
 
+    <style name="Application.Glia.FilePreview.Activity" parent="Application.Glia.Activity.Style">
```
Comment (mustafa-ogun-ozturk) [inline widgetssdk/src/main/res/values/themes.xml:48]: is this needed?

File: widgetssdk/src/main/java/com/glia/widgets/call/CallController.java [hunk 3]
```diff
@@ -166,6 +166,7 @@ public CallController(
         this.isShowEnableCallNotificationChannelDialogUseCase = isShowEnableCallNotificationChannelDialogUseCase;
     }
 
+
```
Comment (mustafa-ogun-ozturk) [inline widgetssdk/src/main/java/com/glia/widgets/call/CallController.java:169]: extra line

File: widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.java [hunk 4]
```diff
@@ -274,6 +254,27 @@ public ChatController(MediaUpgradeOfferRepository mediaUpgradeOfferRepository,
...
+        @Override
+        public void update(Observable observable, Object o) {
+            if (viewCallback != null) {
+                viewCallback.emitUploadAttachments(getFileAttachmentsUseCase.execute());
+                emitViewState(chatState.setShowSendButton(isShowSendButtonUseCase.execute(chatState.lastTypedText)));
+            }
+        }
+    };
+
+    private final String TAG = "ChatController";
```
Comment (mustafa-ogun-ozturk) [inline widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.java:275]: variables can be placed on top of the class

File: widgetssdk/src/main/java/com/glia/widgets/di/Dependencies.java [hunk 5]
```diff
@@ -21,9 +22,11 @@
     private static ControllerFactory controllerFactory;
     private static final List<String> activitiesInBackstack = new ArrayList<>();
     private static INotificationManager notificationManager;
+    private static DownloadsFolderDataSource downloadsFolderDataSource;
```
Comment (mustafa-ogun-ozturk) [inline widgetssdk/src/main/java/com/glia/widgets/di/Dependencies.java:25]: does it have to be static?

File: widgetssdk/src/main/java/com/glia/widgets/filepreview/data/source/local/DownloadsFolderDataSource.java [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+                                @Override
+                                public void onPrepareLoad(Drawable placeHolderDrawable) {
+
+                                }
+                            });
+                }
+        );
+    }
+
+    public Completable putImageToDownloads(String fileName, Bitmap bitmap) {
```
Comment (mustafa-ogun-ozturk) [inline widgetssdk/src/main/java/com/glia/widgets/filepreview/data/source/local/DownloadsFolderDataSource.java:61]: function body can be divided in to different functions

File: widgetssdk/src/main/java/com/glia/widgets/filepreview/ui/State.java [hunk 7]
```diff
@@ -0,0 +1,240 @@
+package com.glia.widgets.filepreview.ui;
+
+import android.graphics.Bitmap;
+
+class State {
```
Comment (mustafa-ogun-ozturk) [inline widgetssdk/src/main/java/com/glia/widgets/filepreview/ui/State.java:5]: Name can be improved

**Review by mustafa-ogun-ozturk — APPROVED** [review]

File: widgetssdk/src/main/res/values/themes.xml (hunk 2 above)
Comment (deniss-mc) [reply widgetssdk/src/main/res/values/themes.xml:48]: yeah - ill use Chat Style instead

File: widgetssdk/src/main/java/com/glia/widgets/filepreview/ui/State.java (hunk 7 above)
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/filepreview/ui/State.java:5]: its package locked, what do you mean by imported?

File: widgetssdk/src/main/java/com/glia/widgets/di/Dependencies.java (hunk 5 above)
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/di/Dependencies.java:25]: fixed

File: widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.java (hunk 4 above)
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.java:275]: fixed

File: widgetssdk/src/main/java/com/glia/widgets/filepreview/domain/usecase/PutImageFileToDownloadsUseCase.java (hunk 1 above)
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/filepreview/domain/usecase/PutImageFileToDownloadsUseCase.java:9]: fixed

File: widgetssdk/src/main/java/com/glia/widgets/filepreview/data/source/local/DownloadsFolderDataSource.java (hunk 6 above)
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/filepreview/data/source/local/DownloadsFolderDataSource.java:61]: will not touch it atm - will separate to API related methods later

File: widgetssdk/src/main/java/com/glia/widgets/filepreview/ui/State.java (hunk 7 above)
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/filepreview/ui/State.java:5]: https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html

**Pull request merged**